java-checks/src/main/java/org/sonar/java/symexecengine/ExecutionState.java

```diff
@@ -36,18 +36,20 @@
 
 public class ExecutionState {
 
-  final ExecutionState parent;
-  private SetMultimap<Symbol, SymbolicValue> reachableValues = HashMultimap.create();
+  public final ExecutionState parent;
+  private SetMultimap<Symbol, SymbolicValue> allReachableValues = HashMultimap.create();
+  private SetMultimap<Symbol, SymbolicValue> alwaysReachableValues = HashMultimap.create();
   private SetMultimap<Symbol, SymbolicValue> unreachableValues = HashMultimap.create();
   /**
    * List of symbol that were declared within this execution state.
    */
+  // FIXME(merciesa): this rather relates to scope and should eventually be moved to the visitor.
```

Comment:
![New](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/new.png) ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)